pkgs/build-support/ocaml/dune.nix

```
@@ -0,0 +1,36 @@
+{ stdenv, fetchurl, ocaml, findlib, dune }:
+
+{ pname, version, buildInputs ? [], ... }@args:
+
+if args ? minimumOCamlVersion &&
+   ! stdenv.lib.versionAtLeast ocaml.version args.minimumOCamlVersion
+then throw "${pname}-${version} is not available for OCaml ${ocaml.version}"
+else
+
+stdenv.mkDerivation ({
+
+  buildPhase = ''
+    runHook preBuild
+    dune build -p ${pname}
+    runHook postBuild
+  '';
+  checkPhase = ''
+    runHook preCheck
+    dune runtest -p ${pname}
+    runHook postCheck
+  '';
+  installPhase = ''
+    runHook preInstall
+    ${dune.installPhase}
```

Just using `opaline` here directly is probably easier to read while
not adding significant complexity. It would also make it possible
to override it.
Would it make sense to add `opaline` in `nativeBuildInputs` as well?

```suggestion
    ${opaline}/bin/opaline -prefix $out -libdir $OCAMLFIND_DESTDIR
```

---

No opposition to this, except that if this is added in the `nativeBuildInputs`, this is something else that won't be overridable (it will be there even if it's not used).

---

ok. I toughed it was always needed.

---

Well, it is always needed with the current installPhase setup, but if you decide to override this phase to use e.g. opam-install itself (as was done not so long ago), then it is not needed anymore.